sdk/nodejs/output.ts

```diff
@@ -534,6 +534,20 @@ export function secret<T>(val: Input<T | undefined>): Output<Unwrap<T | undefine
         o.isKnown, Promise.resolve(true), o.allResources!());
 }
 
+/**
+ * [unsecret] behaves the same as [output] except the returned output takes the existing output and unwraps the secret
+ */
+export function unsecret<T>(val: Output<T>): Output<T> {
+   return new Output(
+        val.resources(), val.promise(/*withUnknowns*/ true),
+        val.isKnown, Promise.resolve(false), val.allResources!());
+}
+
+export function isSecret<T>(val: Output<T>): Promise<boolean> {
+    return val.isSecret.then(x => x);
```

Comment:
We likely need to implement this like the existing internal `isSecretOutput` function, which properly handles Proxies:

https://github.com/pulumi/pulumi/blob/bd1554d33e45f73c8543bc79252f50bb83a03a1d/sdk/nodejs/output.ts#L388-L394

Then, can we get rid of the existing internal func in favor of the new public func?